src/register-action.ts

```diff
@@ -1,74 +1,82 @@
+            }
+
             // To create campaigns, a namespace is used, which can be the current user's username.
             const currentUser = await getCurrentUser()
-            const namespaceName = currentUser?.username
-            console.log('currentUser', currentUser)
+            if (!currentUser) {
+                throw new Error('No current user')
+            }
+            const namespaceName = currentUser.username
 
-            const match = await sourcegraph.app.activeWindow!.showInputBox({
+            const findString = await sourcegraph.app.activeWindow!.showInputBox({
                 prompt: 'Find all matches of:',
             })
-            if (!match) {
+            if (!findString) {
                 return
             }
 
-            const replacement = await sourcegraph.app.activeWindow!.showInputBox({
+            const replacementString = await sourcegraph.app.activeWindow!.showInputBox({
                 prompt: 'Replace with:',
             })
             // Empty string is a valid replacement, so compare directly with undefined.
-            if (replacement === undefined) {
+            if (replacementString === undefined) {
                 return
             }
 
-            const name = `replace-${slugify(match)}-with-${slugify(replacement)}`
-            const description = `Replace ${escapedMarkdownCode(match)} with ${escapedMarkdownCode(replacement)}`
+            const campaignName = `replace-${slugify(findString)}-with-${slugify(replacementString)}`
+            const description = `Replace ${escapedMarkdownCode(findString)} with ${escapedMarkdownCode(
+                replacementString
+            )}`
 
             let percentage = 0
             const { applyURL, diffStat } = await sourcegraph.app.activeWindow!.withProgress(
                 { title: '**Find-replace**' },
-                async reporter =>
+                reporter =>
                     evaluateAndCreateCampaignSpec(namespaceName, {
-                        name,
+                        name: campaignName,
                         on: [
                             {
-                                repositoriesMatchingQuery: match,
+                                repositoriesMatchingQuery: searchQuery,
                             },
                         ],
                         description,
                         steps: [
                             {
-                                fileFilter,
+                                fileFilter: () => true,
                                 editFile: (path, text) => {
-                                    if (!text.includes(match)) {
+                                    if (!text.includes(findString)) {
+                                        // skip the file by returning null
                                         return null
                                     }
 
                                     percentage += (100 - percentage) / 100
```

Comment:
**suggestion**: Document what's happening with the noop `fileFilter` and `percentage` so we remember what we talked about while pairing when we revisit this